bitcoin/src/crypto/sighash.rs

```diff
@@ -1318,6 +1208,183 @@ impl<E> EncodeSigningDataResult<E> {
     }
 }
 
+/// Error computing a taproot sighash.
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[non_exhaustive]
+pub enum TaprootError {
+    /// Could happen only by using `*_encode_signing_*` methods with custom writers, engines writers
+    /// like the ones used in methods `*_signature_hash` do not error.
+    Io(io::ErrorKind),
+    /// Index out of bounds when accessing transaction input vector.
+    InputIndexOutOfBounds(transaction::IndexOutOfBoundsError),
+    /// Using `SIGHASH_SINGLE` without a "corresponding output" (an output with the same index as
+    /// the input being verified) is a validation failure.
+    SingleWithoutCorrespondingOutput {
+        /// Requested index.
+        index: usize,
+        /// Number of transaction outputs.
+        outputs_size: usize,
+    },
```

Comment:
It'd be useful to make this a separate type too for my PR. I'll push mine and we'll resolve who does what.

Also I've also created `TaprootError` but a bit different.

Reply:
Done, see `SingleMissingOutputError`.